File: FitProjectDroid/app/src/main/java/com/fitproject/droid/ui/components/WorkoutMetricFormat.kt
package com.fitproject.droid.ui.components

import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp

object WorkoutMetricFormat {
    val highlightedMetrics = setOf("Reps", "Rest", "RPE", "Tempo")

    fun isHighlighted(name: String): Boolean = name in highlightedMetrics

    fun fieldWidth(name: String): Dp = when (name) {
        "Reps", "Rest", "RPE", "Tempo" -> 68.dp
        else -> 58.dp
    }

    fun formatTempoDisplay(value: String?): String {
        if (value.isNullOrBlank()) return ""
        val digits = value.filter { it.isDigit() }
        if (digits.isEmpty()) return ""
        return digits.take(3).padStart(3, '0').trimStart('0').ifEmpty { "0" }
    }

    fun sanitizeTempoInput(input: String): String =
        input.filter { it.isDigit() }.take(3)

    fun sanitizeMetricInput(name: String, input: String): String = when (name) {
        "Tempo" -> sanitizeTempoInput(input)
        "Reps", "Rest", "RPE" -> input.filter { it.isDigit() || it == '.' }.take(6)
        else -> input
    }
}

File: FitProjectWin/FitProjectWin/Helpers/WorkoutMetricFormat.cs
using Microsoft.UI.Xaml;

namespace FitProjectWin.Helpers;

internal static class WorkoutMetricFormat
{
    private static readonly HashSet<string> HighlightedMetrics = ["Reps", "Rest", "RPE", "Tempo"];

    public static bool IsHighlighted(string name) => HighlightedMetrics.Contains(name);

    public static double FieldWidth(string name) => 40;

    public static double FieldHeight(string name) => 22;

    public static double FieldFontSize() => 12;

    public static Thickness FieldTextPadding() => new(0, 3, 0, 1);

    public static double RowHeight() => FieldHeight("");

    public static double SetColumnWidth() => 28;

    public static double CompleteColumnWidth() => 22;

    public static string FormatTempoDisplay(string? value)
    {
        if (string.IsNullOrWhiteSpace(value)) return "";
        var digits = new string(value.Where(char.IsDigit).ToArray());
        if (digits.Length == 0) return "";
        var normalized = digits.Length >= 3 ? digits[..3] : digits.PadLeft(3, '0');
        var trimmed = normalized.TrimStart('0');
        return trimmed.Length == 0 ? "0" : trimmed;
    }

    public static string SanitizeTempoInput(string input) =>
        new(input.Where(char.IsDigit).Take(3).ToArray());

    public static string SanitizeMetricInput(string name, string input) => name switch
    {
        "Tempo" => SanitizeTempoInput(input),
        "Reps" or "Rest" or "RPE" => new string(input.Where(c => char.IsDigit(c) || c == '.').Take(6).ToArray()),
        _ => input
    };
}